bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -477,12 +506,20 @@ void ReplicaImp::onMessage<preprocessor::PreProcessResultMsg>(preprocessor::PreP
 
 template <>
 void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(ReplicaAsksToLeaveViewMsg *m) {
+  if (activeExecutions_ > 0) {
```

Comment:
Also, execution is a saperate logic from consensus (it can be done even on a different machine!). so there shouldn't be any problem to run view change protocol while executing the requests. Am I missing something?

Reply:
Same.